the-blue-alliance-ios/View Controllers/Root/PadRootViewController.swift

```diff
@@ -0,0 +1,312 @@
+        viewControllers = [masterNavigationController, emptyNavigationController]
+
+        // splitViewController.preferredDisplayMode = .allVisible
+        // splitViewController.delegate = self
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // TODO: Remove these
+    // MARK: - RootController
+
+    func continueSearch(_ searchText: String) -> Bool {
+        return masterViewController.continueSearch(searchText)
+    }
+
+    func show(event: Event) -> Bool {
+        return masterViewController.show(event: event)
+    }
+
+    func show(team: Team) -> Bool {
+        return masterViewController.show(team: team)
+    }
+
+}
+
+private class PadMasterViewController: ContainerViewController, RootController {
+
+    let fcmTokenProvider: FCMTokenProvider
+    let myTBA: MyTBA
+    let pasteboard: UIPasteboard?
+    let photoLibrary: PHPhotoLibrary?
+    let pushService: PushService
+    let searchService: SearchService
+    let statusService: StatusService
+    let urlOpener: URLOpener
+
+    var searchController: UISearchController!
+
+    init(fcmTokenProvider: FCMTokenProvider, myTBA: MyTBA, pasteboard: UIPasteboard?, photoLibrary: PHPhotoLibrary?, pushService: PushService, searchService: SearchService, statusService: StatusService, urlOpener: URLOpener, persistentContainer: NSPersistentContainer, tbaKit: TBAKit, userDefaults: UserDefaults) {
+        self.fcmTokenProvider = fcmTokenProvider
+        self.myTBA = myTBA
+        self.pasteboard = pasteboard
+        self.photoLibrary = photoLibrary
+        self.pushService = pushService
+        self.searchService = searchService
+        self.statusService = statusService
+        self.urlOpener = urlOpener
+
+        let rootTableViewController = PadRootTableViewController(persistentContainer: persistentContainer, tbaKit: tbaKit, userDefaults: userDefaults)
+
+        super.init(viewControllers: [rootTableViewController], persistentContainer: persistentContainer, tbaKit: tbaKit, userDefaults: userDefaults)
+
+        rootTableViewController.delegate = self
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: - View Lifecycle
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        setupSearchController()
+    }
+
+    // MARK: - RootController
+
+    func continueSearch(_ searchText: String) -> Bool {
+        // Pass
+        return true
+    }
+
+    func show(event: Event) -> Bool {
+        // Pass
+        return true
+    }
+
+    func show(team: Team) -> Bool {
+        // Pass
+        return true
+    }
+    
+}
+
+extension PadMasterViewController: SearchContainer, SearchContainerDelegate, SearchViewControllerDelegate {}
+
+extension PadMasterViewController: PadRootTableViewControllerDelegate {
+
+    func rootTypeSelected(_ rootType: RootType) {
+        switch rootType {
```

Comment:
We can probably take advantage of `RootType.supportsPush` and make this code a little bit more modular